crates/rpc/rpc-builder/src/lib.rs

```diff
@@ -609,9 +618,45 @@ impl RpcServerConfig {
         )));
 
         if let Some(builder) = self.http_server_config {
-            let http_server = builder.build(http_socket_addr).await?;
-            server.http_local_addr = http_server.local_addr().ok();
-            server.http = Some(http_server);
+            let mut cors= None;
+            if let Some(domains) = self.http_cors_domains.as_ref() {
+                match domains.as_str() {
+                    "*" => {
+                        cors = Some(CorsLayer::new()
+                            .allow_methods([Method::GET, Method::POST])
+                            .allow_origin(Any)
+                            .allow_headers(Any));
+                    } 
+                    "" => {}
+                    _ => {
+                        let domains_vec = domains.split(",").collect::<Vec<&str>>();
+
+                        let origins = domains_vec
+                            .into_iter()
+                            .map(|domain| domain.parse().unwrap())
+                            .collect::<Vec<HeaderValue>>();
+
```

Comment:
once  move to function, collect into Result:
```suggestion
                            .collect::<Result<Vec<HeaderValue>,_>();

```